invisible_cities/reco/monitor_functions.py

```diff
@@ -0,0 +1,203 @@
+import glob
+from   collections import defaultdict
+
+import numpy  as np
+import tables as tb
+
+from .. database           import load_db             as dbf
+from .. reco               import histogram_functions as histf
+from .. core               import system_of_units     as units
+
+from .. evm .histos        import HistoManager
+from .. io  .pmaps_io      import load_pmaps
+from .. reco.tbl_functions import get_rwf_vectors
+
+
+def pmap_bins(config_dict):
+    """
+    Generates the binning arrays and label of the monitor plots from the a
+    config dictionary that contains the ranges, number of bins and labels.
+
+    Returns a dictionary with the bins and another with the labels.
+    """
+    var_bins   = {}
+    var_labels = {}
+
+    for k, v in config_dict.items():
+        if   "_bins"   in k: var_bins  [k.replace("_bins"  , "")] = [np.linspace(v[0], v[1], v[2] + 1)]
+        elif "_labels" in k: var_labels[k.replace("_labels", "")] = v
+
+    exception = ['S1_Energy', 'S1_Number', 'S1_Time']
+    bin_sel   = lambda x: ('S2' not in x) and (x not in exception)
+    for param in filter(bin_sel, list(var_bins)):
+        var_bins  ['S1_Energy_' + param] = var_bins  ['S1_Energy'] + var_bins  [param]
+        var_labels['S1_Energy_' + param] = var_labels['S1_Energy'] + var_labels[param]
+    var_bins      ['S1_Time_S1_Energy']  = var_bins  ['S1_Time'] + var_bins  ['S1_Energy']
+    var_labels    ['S1_Time_S1_Energy']  = var_labels['S1_Time'] + var_labels['S1_Energy']
+
+    exception = ['S2_Energy', 'S2_Number', 'S2_Time']
+    bin_sel   = lambda x: ('S1' not in x) and (x not in exception) and ('SiPM' not in x)
+    for param in filter(bin_sel, list(var_bins)):
+        var_bins  ['S2_Energy_' + param]  = var_bins  ['S2_Energy'] + var_bins  [param]
+        var_labels['S2_Energy_' + param]  = var_labels['S2_Energy'] + var_labels[param]
+    var_bins      ['S2_Time_S2_Energy']   = var_bins  ['S2_Time']   + var_bins  ['S2_Energy']
+    var_labels    ['S2_Time_S2_Energy']   = var_labels['S2_Time']   + var_labels['S2_Energy']
+    var_bins      ['S2_Energy_S1_Energy'] = var_bins  ['S2_Energy'] + var_bins  ['S1_Energy']
+    var_labels    ['S2_Energy_S1_Energy'] = var_labels['S2_Energy'] + var_labels['S1_Energy']
+    var_bins      ['S2_XYSiPM']           = var_bins  ['S2_XSiPM']  + var_bins  ['S2_YSiPM']
+    var_labels    ['S2_XYSiPM']           = var_labels['S2_XSiPM']  + var_labels['S2_YSiPM']
+
+    del var_bins['S2_XSiPM']
+    del var_bins['S2_YSiPM']
+
+    return var_bins, var_labels
+
+
+def fill_pmap_var_1d(speaks, var_dict, ptype, DataSiPM=None):
```

Comment:
I didn't realize before, but the peak has its own type, so `ptype` is redundant. Instead of `if ptype == "S2"` you can say `if isinstance(speak, S2)` where `S2` comes from `evm.pmaps`.

Reply:
You can get the string representation with `type(speak).__name__`